src/QsCompiler/CompilationManager/ProjectManager.cs

```diff
@@ -798,20 +798,33 @@ public bool ManagerTask(Uri file, Action<CompilationUnitManager> executeTask, ID
         /// </remarks>
         private readonly Action<string, MessageType>? log;
 
+        /// <summary>
+        /// Used to send telemetry events during processing, if the project is compiled with TELEMETRY defined.
+        /// </summary>
+        /// <remarks>
+        /// May be null!
+        /// </remarks>
+        private readonly Action<string, Dictionary<string, string?>, Dictionary<string, int>>? sendTelemetry;
+
         /// <remarks>
         /// If <paramref name="publishDiagnostics"/> is not null,
         /// it is called whenever diagnostics for the project have changed and are ready for publishing.
         /// <para/>
         /// Any exceptions caught during processing are logged using <paramref name="exceptionLogger"/>.
         /// </remarks>
-        public ProjectManager(Action<Exception>? exceptionLogger, Action<string, MessageType>? log = null, Action<PublishDiagnosticParams>? publishDiagnostics = null)
+        public ProjectManager(
+            Action<Exception>? exceptionLogger,
+            Action<string, MessageType>? log = null,
+            Action<PublishDiagnosticParams>? publishDiagnostics = null,
+            Action<string, Dictionary<string, string?>, Dictionary<string, int>>? sendTelemetry = null)
```

Comment:
I suggest using a well defined delegate here to make it easier to understand the purpose of each input parameter of the sendTelemetry action.
And maybe make this an event instead of a standard field.